packages/analytics-js/src/components/core/Analytics.ts

```diff
@@ -445,6 +446,28 @@ class Analytics implements IAnalytics {
       options: payload.options,
       callback: payload.callback,
     });
+
+    // TODO: Maybe we should alter the behavior to send the ad-block page event
+    // even if the SDK is still loaded. It'll be pushed into the to be processed queue.
+
+    // Send automatic ad blocked page event if adblockers are detected on the page
+    if (
+      state.capabilities.isAdBlocked.value === true &&
+      payload.category !== 'RudderJS-Initiated'
+    ) {
+      this.page(
+        pageArgumentsToCallOptions(
```

Comment:
why you use pageArgumentsToCallOptions? just pass expected PageCallOptions object instead

Reply:
do we need to tranck this once per page call or once pr sdk load?

Reply:
1. Addressed
2. In v1.1, we were doing it for every page call with the assumption that page calls will be made only once.

Reply:
lets keep the same behavior then